src/main/java/com/softserve/demo/service/impl/RegisterServiceImpl.java

```diff
@@ -88,18 +98,55 @@ public ProviderDTO createProvider(final ProviderDTO providerDTO) {
         if (userRepository.existsByUsername(user.getUsername())) {
             throw new AlreadyExistException(USERNAME_EXISTS);
         }
-        if (customerRepository.existsByEmail(providerDTO.getEmail()) && providerRepository.existsByEmail(providerDTO.getEmail())) {
+        if (customerRepository.existsByEmail(providerDTO.getEmail()) || providerRepository.existsByEmail(providerDTO.getEmail())) {
             throw new AlreadyExistException(EMAIL_EXISTS);
         }
         user.setPassword(passwordEncoder.encode(providerDTO.getUserDTO().getPassword()));
         Set<Role> roles = new HashSet<>();
         roles.add(Role.PROVIDER);
         user.setRoles(roles);
+        user.setEmail(providerDTO.getEmail());
+        sendVerificationCode(user);
         userRepository.save(user);
         user = userRepository.findByUsername(user.getUsername());
         Provider provider = providerMapper.providerDTOToProvider(providerDTO);
         provider.setUser(user);
         providerRepository.save(provider);
         return providerMapper.providerToProviderDTO(providerRepository.getByEmail(provider.getEmail()));
     }
+
+    /**
+     * This method verify user by verification code
+     * if user with this code is present in database.
+     *
+     * @param activationCode code by which user will be searched
+     * @return true if user is present in database
+     */
+    @Override
+    @Transactional
+    public boolean verifyUser(String activationCode) {
+        Optional<User> optionalUser = userRepository.findByActivationCode(activationCode);
+        if (!optionalUser.isPresent()) {
+            return false;
```

Comment:
May be exception?

Reply:
Yeah, you are right. I thought about it but I wasn't sure that it would be better. But now I see that it will be better. So I will change it :)